lib/trace_mgr.js

```diff
@@ -0,0 +1,611 @@
+        }
+        else {
+            recFrames.set(errstk[i], false);
+        }
+    }
+
+    if (hasRecFrames) {
+        //Compress any recursive frames
+        var cpos = fpos = 0;
+        while (fpos < errstk.length) {
+            if (recFrames.get(errstk[fpos])) {
+                var recArray = [];
+                var spanpos = fpos;
+                var spanend = errstk.lastIndexOf(errstk[fpos]);
+                while (spanpos <= spanend) {
+                    if (recArray.indexOf(errstk[spanpos]) === -1) {
+                        recArray.push(errstk[spanpos]);
+                    }
+
+                    spanend = Math.max(spanend, errstk.lastIndexOf(errstk[spanpos]));
+                    spanpos++;
+                }
+
+                errstk[cpos] = recArray.sort();
+                fpos = spanend + 1;
+            }
+            else {
+                if (cpos !== fpos) {
+                    errstk[cpos] = errstk[fpos];
+                }
+                fpos++;
+            }
+
+            cpos++;
+        }
+
+        errstk = errstk.slice(0, cpos);
+    }
+
+    var chash = 5381;
+    for (var i = 0; i < errstk.length; ++i) {
+        if (Array.isArray(errstk[i])) {
+            for (var j = 0; j < errstk[i].length; ++j) {
+                chash = djbHash(errstk[i][j], chash);
+            }
+        }
+        else {
+            chash = djbHash(errstk[i], chash);
+        }
+    }
+
+    return {
+        kind: eventKind,
+        fbin: fname,
+        hash: chash,
+        stack: errstk
+    };
+}
+
+/*
+* Check if 2 fuzzy stacks are equal.
+*/
+function eqFuzzyStacks(s1, s2) {
+    if (s1.hash !== s2.hash || s1.stack.length !== s2.stack.length) {
+        return false;
+    }
+
+    for (var i = 0; i < s1.stack.length; ++i) {
+        if (Array.isArray(s1.stack[i]) !== Array.isArray(s2.stack[i])) {
+            return false;
+        }
+
+        if (Array.isArray(s1.stack[i])) {
+            if (s1.stack[i].length !== s2.stack[i].length) {
+                return false;
+            }
+
+            for (var j = 0; j < s1.stack[i].length; ++j) {
+                if (s1.stack[j] !== s2.stack[j]) {
+                    return false;
+                }
+            }
+        }
+        else {
+            if (s1.stack[i] !== s2.stack[i]) {
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+///////////////////////////////
+//Bind handlers
+
+function onExitHandler(code) {
+    emitTrace('emitOnExit', code);
+}
+exports.onExitHandler = onExitHandler;
+
+function onUncaughtExceptionHandler() {
+    emitTrace('emitOnException');
+}
+exports.onUncaughtExceptionHandler = onUncaughtExceptionHandler;
+
+function onSigIntHandler(hasUserHandler) {
+    emitTrace('emitOnSigInt');
+
+    if(!hasUserHandler) {
+        //Really exit without triggering any other events
+        process.reallyExit(0);
+    }
+}
+exports.onSigIntHandler = onSigIntHandler;
+
+reentrantDisable = false;
```

Comment:
Cool stuff in general in this module! It would be great to see some unit tests for this module since a lot of the functions are non-trivial